src/main/java/de/tum/in/www1/artemis/service/AttachmentUnitService.java

```diff
@@ -150,6 +106,9 @@ public AttachmentUnit updateAttachmentUnit(AttachmentUnit existingAttachmentUnit
         prepareAttachmentUnitForClient(savedAttachmentUnit, savedAttachment);
         evictCache(updateFile, savedAttachmentUnit);
 
+        slideRepository.deleteAll(existingAttachmentUnit.getSlides());
+        slideSplitterService.splitAttachmentUnitIntoSingleSlides(savedAttachmentUnit);
```

Comment:
Are you sure these calls are always required? Otherwise, please use a conditional to only do database calls when necessary. I'm not sure, for example if the database call on an empty set still gets executed or not.

Reply:
So whenever an unit is updated old slides need to be all deleted and then split the new unit into images again. So i would say so but i added an check if there is no slide at all then do not make delete all call.

Reply:
I mentioned this because, as far as I understand it, there is no reason to not have a non-parsable Attachment in an attachment unit. Like a source file. A Word document. Anything other than a PDF (and maybe images).

Reply:
These calls should be called only when user provides file to update. So I have a check now `if (updateFile != null) {`